include/boost/wintls/certificate.hpp

```diff
@@ -0,0 +1,280 @@
+//
+// Copyright (c) 2021 Kasper Laudrup (laudrup at stacktrace dot dk)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_WINTLS_CERTIFICATE_HPP
+#define BOOST_WINTLS_CERTIFICATE_HPP
+
+#include <boost/wintls/error.hpp>
+#include <boost/wintls/file_format.hpp>
+
+#include <boost/wintls/detail/win32_crypto.hpp>
+
+#include <memory>
+
+namespace boost {
+namespace wintls {
+
+namespace detail {
+struct crypt_context {
+  crypt_context(const std::string& name) {
+    if (!CryptAcquireContextA(&ptr, name.c_str(), nullptr, PROV_RSA_FULL, CRYPT_NEWKEYSET | CRYPT_SILENT)) {
+      detail::throw_last_error("CryptAcquireContextA");
+    }
+  }
+
+  ~crypt_context() {
+    CryptReleaseContext(ptr, 0);
+  }
+
+  HCRYPTPROV ptr = 0;
+};
+
+struct crypt_key {
+  ~crypt_key() {
+    CryptDestroyKey(ptr);
+  }
+
+  HCRYPTKEY ptr = 0;
```

Comment:
I know this is an internal helper class, but I don't like having a public member that is a managed resource.
I would make it private and create a constructor or similar to set it.
I'm not sure but you might be able to use a
<code>unique_ptr<HCRYPTKEY, CryptDestroyKey> </code>
Instead of this class.